component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java

```diff
@@ -454,6 +463,69 @@ private void assertType(final Schema.Type actual, final Schema.Type expected) {
             }
         }
 
+        @Override
+        public Builder before(final String entryName) {
+            orderOverride = true;
+            orderState = Order.BEFORE;
+            orderTarget = entryName;
+            return this;
+        }
+
+        @Override
+        public Builder after(final String entryName) {
+            orderOverride = true;
+            orderState = Order.AFTER;
+            orderTarget = entryName;
+            return this;
+        }
+
+        private enum Order {
+            BEFORE,
+            AFTER,
+            LAST;
+        }
+
+        private Order orderState = Order.LAST;
+
+        private String orderTarget = "";
+
+        // flag if providedSchema's entriesOrder was altered
+        private Boolean orderOverride = Boolean.FALSE;
+
+        private List<String> entriesOrder = new ArrayList<>();
+
+        private void updateOrderState(final String name) {
```

Comment:
Could you put this in external class, something like :
```java
class NamePosition {
    enum Order { BEFORE, AFTER, LAST;  }
    private final Order orderState;
     private final String orderTarget;

     public void updateOrderState(final String newName, List<String> names) { ... }
}
```
(Not mandatory,  but easier to test and to read.)

Reply:
done